### PR TITLE
test-configs.yaml: Add xilinx-zcu102 device

### PR DESCRIPTION
**Pull request by aliceinwire**
Signed-off-by: Alice Ferrazzi <alice.ferrazzi@miraclelinux.com>

### CODE REVIEW

File: config/core/test-configs.yaml [hunk 1]
```diff
@@ -1640,6 +1640,14 @@ device_types:
             - 'tinyconfig'
             - 'kvm_guest.config'
 
+  xilinx-zcu102:
+    mach: zynq
+    class: arm64-dtb
+    boot_method: uboot
+    filters:
+      - blocklist: *allmodconfig_filter
+      - blocklist: {defconfig: ['multi_v7_defconfig+linaro-base+distribution']}
```
Comment (gctucker) [inline config/core/test-configs.yaml:1649]: Where is `+linaro-base+distribution` coming from? As far as I know there aren't any fragment called like that in build-configs.yaml.

Comment (gctucker) [reply config/core/test-configs.yaml:1649]: Ah fair enough, I guess these fragments were added there as well by mistake.  Could you please drop it from `xilinx-zcu102` and maybe also add a patch to drop it from `zynq-zc702`?

File: config/core/test-configs.yaml [hunk 2]
```diff
@@ -1640,6 +1640,11 @@ device_types:
             - 'tinyconfig'
             - 'kvm_guest.config'
 
+  xilinx-zcu102:
```
Comment (gctucker) [inline config/core/test-configs.yaml:1643]: This device type doesn't appear to be in any lab.  However it is defined in LAVA upstream.

There is this instead in lab-cip:
https://lava.ciplatform.org/scheduler/device_type/zynqmp-zcu102

Is this the same device type but with a different name?

**Review by gctucker — COMMENTED** [review]
There's no results on staging, we need to understand why.

File: config/core/test-configs.yaml (hunk 2 above)
Comment (aliceinwire) [reply config/core/test-configs.yaml:1643]: yes is this one.

Comment (aliceinwire) [reply config/core/test-configs.yaml]: fixed the name

**Comment by gctucker** [comment]
This device type is already in upstream LAVA:
https://git.lavasoftware.org/lava/lava/-/blob/master/etc/dispatcher-config/device-types/xilinx-zcu102.jinja2

Waiting for staging results before merging.

**Comment by gctucker** [comment]
@aliceinwire The reason why jobs aren't getting run on this device is because the device type name doesn't match any existing device tree name.  Here's the list of dtb files from an arm64 defconfig build:
```
                "xilinx/zynqmp-zcu102-rev1.0.dtb",
                "xilinx/zynqmp-zcu102-rev1.1.dtb",
                "xilinx/zynqmp-zcu102-revA.dtb",
                "xilinx/zynqmp-zcu102-revB.dtb",
```

Which revision is the device in the CIP lab?

Once we know this, the issue can be solved either by renaming the device type to match the device tree name (and that should be also fixed upstream in LAVA), or the `dtb` name could be explicitly defined in test-configs.yaml like in this example:

```yaml
  odroid-xu3:
    mach: exynos
    class: arm-dtb
    boot_method: uboot
    dtb: 'exynos5422-odroidxu3.dtb'
```

**Comment by aliceinwire** [comment]
Tested and works https://staging.kernelci.org/soc/zynqmp/
we can close this issue

**Comment by gctucker** [comment]
Yep, only the 16K pages flavour isn't booting:
https://staging.kernelci.org/soc/zynqmp/job/next/kernel/next-20210910/plan/baseline/

But that's not a blocker for this PR, other platforms are failing too.  It would be a bit neater to either add a filter to not run the 16K page build on this platform if it's known to not work, or report it so someone can fix it in the kernel.

If you don't think a filter is necessary then this can be merged now.

**Comment by gctucker** [comment]
Also it may be worth adding something to the commit message, to explain what this device is and which tests are being enabled on it.

**Comment by aliceinwire** [comment]
> Also it may be worth adding something to the commit message, to explain what this device is and which tests are being enabled on it.

done. this pull request canbe merged

**Review by gctucker — APPROVED** [review]

**Pull request merged**